lib/src/commands/docs.dart

```diff
@@ -13,123 +14,80 @@ final docs = ChatGroup(
+        final paginator = await pagination.builders(_getPaginationBuilders(searchResults, query, package));
 
-                return EmbedBuilder()
-                  ..color = color
-                  ..title = 'Search results - $query'
-                  ..addField(
-                    name:
-                        'Results in ${package != null ? 'package ${package.packageName}' : 'all packages'}',
-                    content: entry.value.join('\n'),
-                  )
-                  ..addFooter((footer) {
-                    footer.text = 'Page ${entry.key + 1} of $pageCount';
-                  });
-              })
-              .toList(),
-        );
-
-        await context.respond(paginator.initMessageBuilder());
+        await context.respond(paginator);
       }),
     ),
   ],
 );
 
-/// Search autocomplete, but only include elements from a given package (if there is one selected).
-Iterable<ArgChoiceBuilder> autocompleteQueryWithPackage(
-    AutocompleteContext context) {
-  final selectedPackageName = context.interactionEvent.options
-      .cast<
-          IInteractionOption?>() // Cast to IInteractionOption? so we can return `null` in orElse
-      .firstWhere((element) => element?.name == 'package', orElse: () => null)
-      ?.value
-      ?.toString();
+List<MessageBuilder> _getPaginationBuilders(Iterable<DocEntry> searchResults, String query, PackageDocs? package) {
+  var pageCount = 1;
+
+  final foldedResults = searchResults.fold<List<List<String>>>([[]], (pages, entry) {
+    final entryContent = '[${entry.displayName} ${entry.type}](${entry.urlToDocs})';
+
+    // +1 for newline
+    var wouldBeLength = pages.last.join('\n').length + entryContent.length + 1;
+
+    if (wouldBeLength > 1024 || pages.last.length >= 10) {
+      pages.add([]);
+      pageCount++;
+    }
+
+    return pages..last.add(entryContent);
+  });
 
-  PackageDocs? selectedPackage;
-  if (selectedPackageName != null) {
-    selectedPackage = DocsService.instance.getPackageDocs(selectedPackageName);
-  }
+  return foldedResults.asMap().entries.map((entry) {
+    final embed = EmbedBuilder(
+        color: getRandomColor(),
+        title: 'Search results - $query',
+        fields: [
+          EmbedFieldBuilder(
+              name: 'Results in ${package != null ? 'package ${package.packageName}' : 'all packages'}',
+              value: entry.value.join('\n'),
+              isInline: false),
+        ],
+        footer: EmbedFooterBuilder(text: 'Page ${entry.key + 1} of $pageCount'));
 
-  return [
-    // Allow the user to select their current value
-    if (context.currentValue.isNotEmpty)
-      ArgChoiceBuilder(context.currentValue, context.currentValue),
-    ...DocsService.instance
-        .search(context.currentValue, selectedPackage)
-        .take(context.currentValue.isEmpty ? 25 : 24)
-        .map((e) => ArgChoiceBuilder(e.displayName, e.qualifiedName)),
-  ];
+    return MessageBuilder(embeds: [embed]);
+  }).toList();
```

Comment:
Can be replaced with `pagination.splitEmbeds`.

Reply:
how? `/// Create a [MessageBuilder] for a paginated message created by splitting a long [String] into parts and placing the resulting text in [Embed]s.`